packages/flutter_tools/test/general.shard/build_system/targets/asset_transformer_test.dart

```diff
@@ -0,0 +1,377 @@
+        )
+      ],
+    );
+
+    expect(processManager, hasNoRemainingExpectations);
+    expect(logger.errorText, contains(RegExp(
+r'''
+User-defined transformation of asset "asset\.txt" failed.
+Asset transformer my_transformer did not produce an output file\.
+Input file provided to transformer: ".*\.txt"
+Expected output file at: ".*\.txt"
+Full command: Artifact\.engineDartBinary run my_transformer --input=.*\.txt --output=.*\.txt
+stdout:
+
+stderr:
+Transformation failed, but I forgot to exit with a non-zero code\.
+'''
+    )));
+  });
+
+  testWithoutContext('correctly chains transformations when there are multiple of them', () async {
+    final FileSystem fileSystem = MemoryFileSystem();
+    final BufferLogger logger = BufferLogger.test();
+    final Artifacts artifacts = Artifacts.test();
+
+    final File asset = fileSystem.file('asset.txt')
+      ..createSync()
+      ..writeAsStringSync('ABC');
+    const String outputPath = 'output.txt';
+
+    final String dartBinaryPath = artifacts.getArtifactPath(Artifact.engineDartBinary);
+    final FakeProcessManager processManager = FakeProcessManager.list(<FakeCommand>[
+      FakeCommand(
+        command: <Pattern>[
+          dartBinaryPath,
+          'run',
+          'my_lowercase_transformer',
+          RegExp('--input=.*'),
+          RegExp('--output=.*'),
+        ],
+        onRun: (List<String> args) {
+          final ArgResults parsedArgs = (ArgParser()
+              ..addOption('input')
+              ..addOption('output'))
+            .parse(args);
+
+          final String inputFileContents = fileSystem.file(parsedArgs['input']).readAsStringSync();
+          fileSystem.file(parsedArgs['output'])
+            ..createSync()
+            ..writeAsStringSync(inputFileContents.toLowerCase());
+        },
+      ),
+      FakeCommand(
+        command: <Pattern>[
+          dartBinaryPath,
+          'run',
+          'my_distance_from_ascii_a_transformer',
+          RegExp('--input=.*'),
+          RegExp('--output=.*'),
+        ],
+        onRun: (List<String> args) {
+          final ArgResults parsedArgs = (ArgParser()
+              ..addOption('input')
+              ..addOption('output'))
+            .parse(args);
+
+          final String inputFileContents = fileSystem.file(parsedArgs['input']).readAsStringSync();
+          final StringBuffer outputContents = StringBuffer();
+
+          for (int i = 0; i < inputFileContents.length; i++) {
+            outputContents.write(inputFileContents.codeUnitAt(i) - 'a'.codeUnits.first);
+          }
+
+          fileSystem.file(parsedArgs['output'])
+            ..createSync()
+            ..writeAsStringSync(outputContents.toString());
+        },
+      ),
+    ]);
+
+    final AssetTransformer transformer = AssetTransformer(
+      processManager: processManager,
+      logger: logger,
+      fileSystem: fileSystem,
+      dartBinaryPath: dartBinaryPath,
+    );
+
+    await transformer.transformAsset(
```

Comment:
let's expect on the return value

Reply:
Addressed in [expect on transformationResult](https://github.com/flutter/flutter/pull/143815/commits/7a9708ec4d120bca184689574016098ad0164fcd)